mm2src/coins/lightning/ln_utils.rs

```diff
@@ -336,8 +514,280 @@ async fn ln_p2p_loop(ctx: MmArc, peer_manager: Arc<PeerManager>, listener: TcpLi
     }
 }
 
+#[cfg(not(target_arch = "wasm32"))]
+struct ConfirmedTransactionInfo {
+    txid: Txid,
+    header: BlockHeader,
+    index: usize,
+    transaction: Transaction,
+    height: u32,
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+impl ConfirmedTransactionInfo {
+    fn new(txid: Txid, header: BlockHeader, index: usize, transaction: Transaction, height: u32) -> Self {
+        ConfirmedTransactionInfo {
+            txid,
+            header,
+            index,
+            transaction,
+            height,
+        }
+    }
+}
+
+// Used to order 2 transactions if one spends the other by the spent transaction first
+#[cfg(not(target_arch = "wasm32"))]
+fn cmp_txs_for_spending(spent_tx: &Transaction, spending_tx: &Transaction) -> Ordering {
+    for tx_in in &spending_tx.input {
+        if spent_tx.txid() == tx_in.previous_output.txid {
+            return Ordering::Less;
+        }
+    }
+    Ordering::Equal
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+async fn process_tx_for_unconfirmation(txid: Txid, filter: Arc<PlatformFields>, channel_manager: Arc<ChannelManager>) {
+    if let Err(err) = filter
+        .platform_coin
+        .as_ref()
+        .rpc_client
+        .get_transaction_bytes(H256::from(txid.as_hash().into_inner()).reversed())
+        .compat()
+        .await
+        .map_err(|e| e.into_inner())
+    {
+        if let UtxoRpcError::ResponseParseError(ref json_err) = err {
```

Comment:
`ResponseParseError` doesn't always mean that transaction is not found. It might be an unsupported Electrum server version. Is it possible to handle this case more precisely?

Reply:
Done. I used the following errors:

- Error Code: https://github.com/spesmilo/electrumx/blob/0e321d2195b0b1536f2ffeb6f9181f3e3d5b8b0a/electrumx/server/session.py#L49
- Error Message: https://github.com/spesmilo/electrumx/blob/0e321d2195b0b1536f2ffeb6f9181f3e3d5b8b0a/electrumx/server/session.py#L739
- `getrawtransaction` Error: https://github.com/bitcoin/bitcoin/blob/dca9ab48b80ff3a7dbe0ae26964a58e70d570618/src/rpc/rawtransaction.cpp#L241-L243
- `getrawtransaction` Error Code: https://github.com/bitcoin/bitcoin/blob/dca9ab48b80ff3a7dbe0ae26964a58e70d570618/src/rpc/protocol.h#L41

Hope I didn't miss something 🙂